### DISCUSSION TITLE
pb for postgresql connection : Waiting for the server to start

### DISCUSSION THREAD

**Discussion by NefissaA** [leokhoa/laragon#533] [General]
I use laragon for web developpement, and I need to send data form to postgresql database. I installed postgresql V12 / V14 and V15 and  I add it to laragon : menu/tools/quick add and then I updated config file with the link to binary file, when I started laragon, an error message dispayed "Waiting for the server to start", how can I start the server ? this pb is only with postgres, and in the interface of laragon this message is in red "PostgreSQL postgresql-12.15-2 started". thank you for help

**Comment by Pen-y-Fan** [comment]
The first thing I would check is permissions, see my blog [How to set up Laragon on a new Windows computer (part 2) - permissions](https://pen-y-fan.github.io/2022/11/08/how-to-set-up-laragon-on-a-new-windows-computer-part-2/). I recommend a standard user account be given permission to modify the Laragon folder, this allows data files to be created.

**Comment by alphaelf** [comment]
for me `pgsql v15.1` run perfectly but `v16.2` get me same error: `Waiting for the server to start`

**Comment by bloodykheeng** [comment]
@NefissaA did u find a solution am facing same problem with v 17

**Comment by bloodykheeng** [comment]
hello guys this is what i did to make mine working
---

I was having trouble installing **PostgreSQL** on Windows — the installation kept failing, and I couldn’t figure out why.

Later, I suspected something was already using the default PostgreSQL port, **5432**. To check this, I opened **Command Prompt as Administrator** (this is important for certain commands to work properly) and ran:

```bash
netstat -aon | findstr :5432
```

Output:

```
C:\Users\BK>netstat -aon | findstr :5432
  TCP    0.0.0.0:5432           0.0.0.0:0              LISTENING       6176
  TCP    [::]:5432              [::]:0                 LISTENING       6176
```
> This shows that **port 5432 is in use by a process with PID `6176`**, listening on all network interfaces.

To identify what was using that PID, I ran:

```bash
tasklist /FI "PID eq 6176"
```

Which returned:

```
Image Name                     PID         Session Name        Mem Usage
=========================     ========    ================    ============
postgres.exe                  6176        Console             25,000 K
```

So PostgreSQL was already running in the background and using port 5432.

To fix the issue, I did the following:

1. **Opened Command Prompt as Administrator** (⚠️ required to kill the process successfully).
2. Stopped the process using:

   ```bash
   taskkill /PID 6176 /F
   ```
3. Deleted the existing PostgreSQL data folder to clean up any broken or incomplete setup:

   ```
   C:\laragon\data\postgresql
   ```
4. Restarted **Laragon**, and everything worked perfectly — PostgreSQL started without issues.

---

**Comment by FadhilPrawira** [comment]
This is what i did
Laragon version 6.0.220916
Windows 11 Pro 25H2

What i did: change from Postgresql 18.0 (https://get.enterprisedb.com/postgresql/postgresql-18.0-2-windows-x64-binaries.zip) to 18.1 (https://get.enterprisedb.com/postgresql/postgresql-18.1-1-windows-x64-binaries.zip)

I replace the source url in `C:\laragon\usr\packages.conf` so it looks like this
postgresql-18=https://get.enterprisedb.com/postgresql/postgresql-18.1-1-windows-x64-binaries.zip

Then i delete the postgresql 18.0 and run postgresql 18.1. I got error `Waiting for the server to start`

I run @bloodykheeng solution https://github.com/leokhoa/laragon/discussions/533#discussioncomment-13645912 but i only find ADB is using 5432 port (which is weird). Then i restart my computer. 

Later i notice my folder in `C:\laragon\data` already have `postgresql-18` folder but with different date. Without hesitation, i delete that folder and run my `postgresql-18.1`, and its work!
